stix_shifter_modules/guardium/stix_transmission/api_client.py

```diff
@@ -1,250 +1,64 @@
-            raise IOError(
-                    3001, "Guardium Credential not provided in the connection / configuration object")
+    def ping_data_source(self):
+        # Pings the data source
+        if self.client.validate_response(self.client.request_token(), "", False):
+            return {"code": 200, "success": True}
         else:
-            self.credential = {"username": username, "password": password, "grant_type": grant_type,
-                               "client_id": client_id, "client_secret": client_secret}
-
-        host_port = connection.get('host') + ':' + \
-            str(connection.get('port', ''))
-
-        url_modifier_function = None
-        proxy = connection.get('proxy')
-        if proxy is not None:
-            proxy_url = proxy.get('url')
-            proxy_auth = proxy.get('auth')
-            if (proxy_url is not None and proxy_auth is not None):
-                self.headers['proxy'] = proxy_url
-                self.headers['proxy-authorization'] = 'Basic ' + proxy_auth
-            if proxy.get('x_forward_proxy', None) is not None:
-                self.headers['x-forward-url'] = 'https://' + \
-                    host_port + '/'  # + endpoint, is set by 'add_endpoint_to_url_header'
-                host_port = proxy.get('x_forward_proxy')
-                if proxy.get('x_forward_proxy_auth', None) is not None:
-                    self.headers['x-forward-auth'] = proxy.get(
-                        'x_forward_proxy_auth')
-                self.headers['user-agent'] = 'UDS'
-                url_modifier_function = self.add_endpoint_to_url_header
-
-        self.client = RestApiClient(host_port,
-                                    None,
-                                    self.headers,
-                                    url_modifier_function,
-                                    cert_verify=connection.get(
-                                        'selfSignedCert', True),
-                                    sni=connection.get('sni', None)
-                                    )
-        self.search_timeout = connection['options'].get('timeout')
+            return {"success": False}
 
-    def add_endpoint_to_url_header(self, url, endpoint, headers):
-        # this function is called from 'call_api' with proxy forwarding,
-        # it concatenates the endpoint to the header containing the url.
-        headers['x-forward-url'] += endpoint
-        # url is returned since it points to the proxy for initial call
-        return url
-
-    def add_header_elements(self, key, value):
-        self.headers[key] = value
-        return
-
-    def ping_box(self):
-        # Subroto -- Guardium does not have ping facility
-        # We test if we can get the access token if we can then success = true
-        #
+    def get_search_status(self, search_id):
+        # Check the current status of the search
+        return {"code": 200, "status": "COMPLETED"}
+    
+    def get_status(self, search_id):
```

Comment:
already mention that in the older pr. this is wrong way to return status of the search. you are calling `self.client.access_token` to get the status which is similar to ping. 

as a synchronous connector you don't receive any status so just return the hard coded status just like `get_search_status`. I guess you can just use `get_search_status()` function in `status_connector.py`

Reply:
get_search_status() function in status_connector.py
calls my api_client method.
Ill change get_search_status to call get_status which.

Reply:
done